extendedstats/windows.go

```diff
@@ -67,20 +67,47 @@ type TCPINFO_V1 struct {
 	SndLimBytesSnd  uint64
 }
 
-func (es *ExtendedStats) IncorporateConnectionStats(rawConn net.Conn) error {
-	tlsConn, ok := rawConn.(*tls.Conn)
-	if !ok {
-		return fmt.Errorf(
-			"OOPS: Could not get the TCP info for the connection (not a TLS connection)",
-		)
-	}
-	tcpConn, ok := tlsConn.NetConn().(*net.TCPConn)
-	if !ok {
-		return fmt.Errorf(
-			"OOPS: Could not get the TCP info for the connection (not a TCP connection)",
-		)
-	}
-	if info, err := getTCPInfo(tcpConn); err != nil {
+// https://pkg.go.dev/golang.org/x/sys/unix#TCPInfo
+// Used to allow access to TCPInfo in like manner to unix.
+type TCPInfo struct {
```

Comment:
Per our discord conversation I agree that this is the best of the worst solutions. Once we pull in this patch, I propose that we move this out of platform-specific files so that we don't have to keep it in sync? Thoughts?

Reply:
In the sense that we add our own TCPInfo struct (with necessary stats) in rpm.go and then fill it in each individual platform extendedstats?

Reply:
Exactamundo!